### ISSUE TITLE
Entity type classification for retrieval ranking

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#552]
## Problem

After implementing virtual predicates (#549) and compound preservation (#550), retrieval may still struggle to prioritize semantically relevant matches.

**Example:**
- Query: "Kiu fondis Esperanton?" (Who founded Esperanto?)
- Wants: PERSON founding LANGUAGE
- May retrieve: ORGANIZATION founding ORGANIZATION

Even with correct predicate structure, we need to weight matches by entity type relevance.

## Solution

Train entity type classifier to categorize entities, then use types for relevance weighting.

**Entity types:**
- PERSON (zamenhof, chavet)
- LANGUAGE (esperant)
- ORGANIZATION (rond, lig, asocio)
- PLACE (urb, land)
- EVENT (kongres, milit)
- ABSTRACT (ide, sci)

**Architecture:**
```python
class EntityTypeClassifier(nn.Module):
    def __init__(self, compositional_embedding, num_types=10):
        self.comp_emb = compositional_embedding  # Use existing 128D embeddings
        self.classifier = nn.Linear(128, num_types)
    
    def forward(self, word_struct):
        emb = self.comp_emb.encode_word(**word_struct)
        return self.classifier(emb)  # Type probabilities
```

**Training data sources:**
1. Suffix patterns (deterministic):
   - -ul = PERSON (esperantul)
   - -ism = ABSTRACT (kapital-ism)
   - -ej = PLACE (libr-ej)
2. Wikipedia entity mentions (silver labels)
3. Manual annotation (gold labels, small set)

**Use in retrieval:**
```python
# Score predicate match by entity type alignment
query_wants = {"subject": PERSON, "object": LANGUAGE}
predicate_has = {"subject": entity_type("zamenhof"), "object": entity_type("esperant")}

type_alignment_score = cosine(query_wants, predicate_has)
final_score = predicate_match_score * type_alignment_score
```

## Implementation

**New files:**
- `klareco/models/entity_type_classifier.py`
- `scripts/train_entity_classifier.py`
- `scripts/prepare_entity_type_data.py`

**Integration:**
- `klareco/rag/kuzu_inverted_index.py` - Add type-based weighting

**Parser changes**: NONE  
**AST changes**: Optional `entity_type` annotation  
**Training needed**: YES - new model (~10K labeled examples)  
**Expected impact**: +10-20% exact accuracy

## Success Criteria

- [ ] Classifier achieves >85% accuracy on held-out entity types
- [ ] Suffix-based features (deterministic) provide baseline
- [ ] Integration with retrieval improves ranking
- [ ] Re-run 50-question eval shows improvement

## Priority

**P2** - Implement AFTER #549 and #550. Evaluate whether type classification is needed based on results.

If virtual predicates + compound preservation get to 60%+ exact accuracy, this may not be necessary. Defer until error analysis shows type confusion is the bottleneck.

**Comment by marctjones** [comment]
## Implementation Priority: Lower (P2)

This is a nice-to-have enhancement after deterministic improvements prove their limits.

**Why P2:**
- Deterministic entity detection + boosting (#571) should work well first
- Only train this if deterministic methods show clear gaps
- Small model (500K params) but requires training data generation

## Dependencies
- After: #549, #554, #568 (deterministic improvements)
- Benefits: #571 (dynamic entity boosting can use type predictions)
- Benefits: #555 (better ranking with semantic types)

## Implementation Order
1. Phases 1-2: All deterministic improvements
2. Evaluate: Measure remaining failures
3. If semantic type confusion causes >10% errors, then implement this

**Justification:** Stay true to "deterministic first" philosophy. Only add learned parameters when deterministic methods clearly insufficient.

**Estimated time:** 1 week
**Params:** ~500K (total: 320K + 180K + 500K = 1M)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
